solr/solrj/src/java/org/apache/solr/client/solrj/impl/Http2SolrClient.java

```diff
@@ -136,6 +136,11 @@ public class Http2SolrClient extends SolrClient {
   private long requestTimeoutMillis;
 
   protected ResponseParser parser = new BinaryResponseParser();
+  private final Set<String> defaultMimeTypes =
```

Comment:
wouldn't it be better if this happens after the parser is initialized from the builder? https://github.com/apache/solr/blob/237ff32720ac02c9d950e1aa863d0c957a0c6702/solr/solrj/src/java/org/apache/solr/client/solrj/impl/Http2SolrClient.java#L191 could cover the case there a different parser is passed in, in case that is useful.

Reply:
Ah, definitely, you are right.